src/security_utilities_rust/src/microsoft_security_utilities_core/identifiable_scans.rs

```diff
@@ -110,11 +138,136 @@ impl PossibleScanMatch {
             ScanMatchType::His32Utf16 => { HIS_32_UTF16_LEN },
             ScanMatchType::His64Utf8 => { HIS_64_UTF8_LEN },
             ScanMatchType::His64Utf16 => { HIS_64_UTF16_LEN },
+            ScanMatchType::His39Utf8 => { HIS_39_UTF8_LEN },
+            ScanMatchType::His39Utf16 => { HIS_39_UTF16_LEN },
+            ScanMatchType::His40Utf8 => { HIS_40_UTF8_LEN },
+            ScanMatchType::His40Utf16 => { HIS_40_UTF16_LEN },
+            ScanMatchType::HisA7Utf8 => { HIS_A7_UTF8_LEN },
+            ScanMatchType::HisA7Utf16 => { HIS_A7_UTF16_LEN },
+            ScanMatchType::HisA8Utf8 => { HIS_A8_UTF8_LEN },
+            ScanMatchType::HisA8Utf16 => { HIS_A8_UTF16_LEN },
         }
     }
 
     pub fn match_type(&self) -> &ScanMatchType { &self.mtype }
 
+    fn his_a7_matched_bytes(data: &[u8]) -> usize {
+        /*
+         * 3 Base64 + 3 signature + 31 Base64 + optional 1 [=]
+         */
+        if data.len() < HIS_A7_UTF8_LEN {
+            return 0;
+        }
+
+        for b in &data[0..3] {
+            if !b.is_url_unreserved() {
+                return 0;
+            }
+        }
+
+        for b in &data[6..37] {
+            if !b.is_url_unreserved() {
+                return 0;
+            }
+        }
+
+        HIS_A7_UTF8_LEN
+    }
+
+    fn his_a8_matched_bytes(data: &[u8]) -> usize {
+        /*
+         * 3 Base64 + 3 signature + 34 Base64 + optional 1 [=]
+         */
+        if data.len() < HIS_A8_UTF8_LEN {
+            return 0;
+        }
+
+        for b in &data[0..3] {
+            if !b.is_url_unreserved() {
+                return 0;
+            }
+        }
+
+        for b in &data[6..40] {
+            if !b.is_url_unreserved() {
+                return 0;
+            }
+        }
+
+        HIS_A8_UTF8_LEN
+    }
+
+    fn his_39_matched_bytes(data: &[u8]) -> usize {
+        /*
+         * 42 Base64 + 4 signature + 1 [A-D] + 5 Base64 + optional 1 [=]
+         */
+        if data.len() < 52 {
+            return 0;
+        }
+
+        for b in &data[0..42] {
+            if !b.is_base64() {
+                return 0;
+            }
+        }
+
+        /* NOTE: We skip the signature since we already checked */
```

Comment:
>* NOTE: We skip the signature since we already checked */ [](http://example.com/codeflow?start=9&length=57)

NIT: this comment is true in other helpers and it may be useful to add it to others, for code readability.

Reply:
specifically, in `his_a7_matched_bytes` and `his_a8_matched_bytes`

Reply:
Added.